### PR TITLE
Make App Engine deployment more re-usable.

### PR DESCRIPTION
**Pull request by oliverchang**
- Move App Engine YAML files to deployment/<project>/gae.
- Modify the App Engine deployment script to symlink in the YAML files
  from the source config directory prior to deployment. This is
  necessary because `gcloud app deploy` requires the yaml files to be
  in the same directory as the application root.
- Also add GAE configs for "oss-vdb-test".

### CODE REVIEW

**Comment by andrewpollock** [comment]
> * gcloud auth deploy

`gcloud app deploy`?

File: deployment/staging.yaml [hunk 1]
```diff
@@ -31,8 +31,7 @@ steps:
   entrypoint: 'bash'
   args: [ '-c', "rm /workspace/service_account.json" ]
 - name: 'gcr.io/oss-vdb/deployment'
-  args: ['bash', '-ex', 'deploy.sh', 'oss-vdb', 'app.yaml', '--no-promote']
-  dir: gcp/appengine
+  args: ['bash', '-ex', 'gcp/appengine/deploy.sh', 'oss-vdb', 'deployment/oss-vdb/gae', 'app.yaml', '--no-promote']
```
Comment (andrewpollock) [inline deployment/staging.yaml:34]: I think the project ID here should be `oss-vdb-staging` based on the filename?

**Comment by andrewpollock** [comment]
> `gcloud app deploy`?

Pretty confident the answer is "yes", so I'll update the description accordingly.

**Review by michaelkedar — APPROVED** [review]

File: deployment/staging.yaml (hunk 1 above)
Comment (oliverchang) [reply deployment/staging.yaml:34]: This is the staging instance within the prod GCP project, so this is intentional.

**Comment by oliverchang** [comment]
> > `gcloud app deploy`?
> 
> Pretty confident the answer is "yes", so I'll update the description accordingly.

Yep thanks !!!

**Pull request merged**